stream_interfaces.go

```diff
@@ -39,8 +59,25 @@ type ServerStreamingServer[Res any] interface {
 // message stream and the type of the unary response message. It is used in
 // generated code.
 type ClientStreamingClient[Req any, Res any] interface {
+	// Send sends a request message to the server. This method is called repeatedly
+	// to send all messages as part of the client’s request stream.
+	// The message type is determined by the Req type parameter
+	// of the ClientStreamingClient receiver. It returns an error
+	// if the message could not be sent.
 	Send(*Req) error
+
+	// CloseAndRecv closes the sending side of the request stream and waits for the server
+	// to send a unary response message. This method is typically called after
+	// sending all request messages to signal the end of the stream and to receive
+	// the final response from the server. The response message type is determined
+	// by the Res type parameter of the ClientStreamingClient receiver.
+	// It returns the received response message and an error if any occurred.
+	// The CloseAndRecv method must be called once and only once to complete the RPC.
 	CloseAndRecv() (*Res, error)
+
+	// ClientStream represents the basic client-side streaming operations.
```

Comment:
No need to write inline comments for parent interface `ClientStream` as the actual interface declaration will already have it